### DISCUSSION TITLE
is `get_raw_config` deprecated?

### DISCUSSION THREAD

**Discussion by devgony** [LazyVim/LazyVim#4978] [Q&A]
I'm configuring java lsp according to [this doc](http://www.lazyvim.org/extras/lang/java#nvim-jdtls).
But it throws `Undefined field get_raw_config` on below line
```lua
      root_dir = LazyVim.lsp.get_raw_config("jdtls").default_config.root_dir,
```

Seems like there is no get_raw_config in lsp.lua

**Comment by folke** [comment]
nope, still exists. If it doesn/t for you then you're not using LAzyVim.
Also, that doc shows the default options used for the extra, so you should **NOT** copy all of that in your own config.

**Reply by devgony** [reply]
@folke 
Oh.. i realized that my LazyVim was not latest. (it was v12).
But even though it seems like properly configured, it still creates setting files on projectRoot not the location i configured.
It seems like this configuration does not work..
I don't know whether lazyvim didn't adjust the configuration yet or jdtls config itself was wrong.
```lua
-- ~/.config/nvim/lua/plugins/jdtls.lua
    ..
    -- Where are the config and workspace dirs for a project?
    jdtls_config_dir = function(project_name)
      return vim.fn.stdpath("cache") .. "/jdtls/" .. project_name .. "/config"
    end,
    jdtls_workspace_dir = function(project_name)
      return vim.fn.stdpath("cache") .. "/jdtls/" .. project_name .. "/workspace"
    end,
```

**Reply by folke** [reply]
no idea. I don't do java

**Reply by micah-svenson** [reply]
You need to pass an additional system property to jdtls directly to prevent metadata files from being generated in the project. 

I posted a solution in another discussion.
https://github.com/LazyVim/LazyVim/discussions/4972#discussioncomment-12606165
